File: beta/Beta - Rotated string.js
// Beta - Rotated string

// const isRotation = (s1, s2) => {
//     let end = start = 0
//     for (let i = 0, j = s1.length - 1; i < s1.length - 1; i++, j--) {
//         start = s1.slice(0, i + 1)
//         end = s1.slice(-j)
//         if (end + start == s2)
//             return true
//     }
//     return s1 == s2
// }

function isRotation(s1, s2) {
    for (let i = 0; i < s1.length; i++)
        if (s2 == s1.slice(-i) + s1.slice(0, -i))
            return true
    return s1 == s2
}

// const isRotation = (s1, s2) => s1.length == s2.length && (s2 + s2).indexOf(s1) > -1
// const isRotation = (s1, s2) => s1.length == s2.length && (s2 + s2).includes(s1)

q = isRotation('hello', 'ohell') // true
q
q = isRotation('hello', 'lloeh') // false
q
q = isRotation('', '') // true
q

File: beta/Beta - Rotated string.py
# Beta - Rotated string

""" Write to function that takes as argument two strings and returns True
if one string is a rotation of the other or else it returns False.

# ohell is left rotation of hello
is_rotation('hello','ohell') => True

# elloh is right rotation of hello 
is_rotation('hello','elloh') => True """


# def is_rotation(s1, s2):
#     for i in range(len(s2)):
#         if s2[i:] + s2[:i] == s1:
#             return True
#     return s1 == s2

def is_rotation(s1, s2):
    return len(s1) == len(s2) and s2 in (s1 * 2)


q = is_rotation('hello', 'ohell'), True
q
q = is_rotation('hello', 'lloeh'), False
q
